cpp/include/rmm/mr/detail/device_memory_resource_view.hpp

```diff
@@ -0,0 +1,189 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/detail/error.hpp>
+#include <rmm/mr/device_memory_resource.hpp>
+
+#include <cuda/memory_resource>
+
+#include <cstddef>
+#include <type_traits>
+
+namespace RMM_NAMESPACE {
+namespace mr::detail {
+
+/**
+ * @brief A copyable view wrapping a `device_memory_resource*` pointer.
+ *
+ * This class serves as a temporary bridge to enable compatibility with CCCL 3.2's memory resource
+ * design, which requires resource types to be copyable but no longer accepts raw pointers directly.
+ * Since `device_memory_resource` is a virtual base class that cannot be copied, this view provides
+ * a copyable wrapper around a `device_memory_resource*` pointer.
+ *
+ * This is an internal implementation detail and should not be used directly by users. It will be
+ * removed once RMM fully migrates away from the `device_memory_resource` virtual base class.
+ *
+ * @note This class does NOT manage the lifetime of the wrapped pointer. The caller is responsible
+ * for ensuring the pointed-to resource remains valid for the lifetime of this view.
+ */
+class device_memory_resource_view {
+ public:
+  /**
+   * @brief Constructs a view wrapping the given `device_memory_resource` pointer.
+   *
+   * @throws rmm::logic_error if `ptr` is null
+   *
+   * @param ptr Non-null pointer to a `device_memory_resource`
+   */
+  device_memory_resource_view()                                              = default;
+  ~device_memory_resource_view()                                             = default;
+  device_memory_resource_view(device_memory_resource_view const&)            = default;
+  device_memory_resource_view(device_memory_resource_view&&)                 = default;
+  device_memory_resource_view& operator=(device_memory_resource_view const&) = default;
+  device_memory_resource_view& operator=(device_memory_resource_view&&)      = default;
+
+  device_memory_resource_view(device_memory_resource* ptr) : resource_ptr_{ptr}
```

Comment:
Since this class only wraps a raw pointer without ownership, it should follow the Rule of Zero. Only this non-null constructor is enough; explicit defaulting of ctor/dtor/assignment operators isn’t required.

Reply:
Okay. I know this class _must_ be copyable but maybe explicit defaults aren't needed. I will check what happens when I compile RMM + cuDF without these.